### ISSUE TITLE
[AGS 4.0] Allow to sort, filter, move, edit(fully) schema properties.

### ISSUE THREAD

**Issue by tarnos12** [adventuregamestudio/ags#1621]
**Describe the problem**
Schema properties cannot be moved once created, after I have removed one of the properties, then added it again, it appears at the very bottom of my schema list which is not ideal as things get messy when you have 20 or more properties and they are scattered around.
Another thing is that it's not possible to edit schema name, so removing and creating it again is the only option.

**Suggested change**
At the very least a way to right click and Move Up/Down option(the easiest solution I guess), the better one would be to have the ability to sort the list(or it should auto sort itself alphabetically), this way as long as I name my properties properly I will never lose them. 
Editing the name of the property would be great too.

**Comment by ericoporto** [comment]
The properties need an order for being written in Game.agf keeping always the same order, so one can compare the Game.agf files in git history. But assuming this order doesn't matter for the user of the Editor, the order presented may always be sorted.

If there's more than a sorting option (Say alphabetical, alphabetical but by types, ...), then the last sorting option has to be saved (where? I don't know).

Editing the name, perhaps can be done in-place, there's code for this in the Project Explorer. In case of a right click edit, then it would reload the property pane but with the information filled.

**Comment by ivan-mogilko** [comment]
There's an option to not save properties sort as a part of game data, but as a part of user (workspace) data.

I cannot tell whether this is preferable, but that's an option to keep in mind, imo.

**Comment by ericoporto** [comment]
> Editing the name of the property would be great too. 
- I think renaming is implemented in: https://github.com/adventuregamestudio/ags/pull/2963
 
> At the very least a way to right click and Move Up/Down option(the easiest solution I guess), the better one would be to have the ability to sort the list(or it should auto sort itself alphabetically) 
- The sorting stuff was done in commit [b52b9ee](https://github.com/adventuregamestudio/ags/commit/b52b9eec72b79afc57177acfe87d5fe96892dc6e) and [458fa37](https://github.com/adventuregamestudio/ags/commit/458fa37f0f6819efb908b3e07dad4189c488508d) (so since 3.6.3.8) 
- I guess moving up or down is not something desirable in this new way of looking at the schema since it is always sorted by something.

> Allow to sort, filter, move, edit(fully) schema properties.
- The filtering mentioned in the title of the issue is not mentioned in the issue body, I guess it may not be as necessary, if desired it would need a single text input box somewhere where one could type to reduce the listed custom properties with some indication that the list shown is filtered and doesn't contain _all_ properties in the schema.

For reference this is how the Custom Properties schema columns look like now. Notice the arrow that says which column is used for ordering, and if the direction is ascending or descending.

<img width="954" height="545" alt="Image" src="https://github.com/user-attachments/assets/41682631-d7fe-4902-bd59-fb2d798bbd4f" />

(the "applies to" looks like chemistry formula!)

I think this issue could be closed?

**Comment by ivan-mogilko** [comment]
Yes, looks like this is completed in AGS 3.6.3, except for the "filter", but it was never elaborated. If still wanted, it may be opened as a separate feature suggestion.
